src/engagement.py

```diff
@@ -0,0 +1,115 @@
+from core_data_modules.util import PhoneNumberUuidTable
+
+from src.lib.pipeline_configuration import PipelineConfiguration
+
+class Engagement(object):
+    @staticmethod
+    def generate(data, phone_number_uuid_table):
+        """
+        prints out engagements numbers per Episode
+        :param data: TracedData objects that have been manually labelled.
+        :type: list of TracedData
+        :param: phone_number_uuid_table: uuid <-> phone number look up table
+        :type phone_number_uuid_table: core_data_modules.util.PhoneNumberUuidTable.
+        :return: data
+        :rtype: list of TracedData
+        """
+        s01e01_uids = set()
+        s01e02_uids = set()
+        s01e03_uids = set()
+        s01e04_uids = set()
+        s01e05_uids = set()
+        s01e06_uids = set()
+        s01e07_uids = set()
+        repeated_all_episodes = set()
+        repeated_any_episodes = set()
+        icraf_all_uids = set()
+
+        for td in data:
+            for plan in PipelineConfiguration.RQA_CODING_PLANS:
+                if plan.raw_field == "rqa_s01e01_raw":
```

Comment:
There is a huge amount of duplicated code here (or if it's different, it's really not obvious where). Can you reduce this e.g. by iterating over all the raw fields in PipelineConfiguration.RQA_CODING_PLANS and storing in a dict of sets?